### PR TITLE
Tags checker test

### PR DESCRIPTION
**Pull request by AlbaMontis**

### CODE REVIEW

File: tests/testranks.js [hunk 1]
```diff
@@ -9,7 +9,7 @@
 /** @const {!RegExp} */ var PATTERN = /^\d([,\.]*\d)*$/;
 
 /** @const {!Object.<string, function(string):boolean>} */ var VALIDATORS = {
-  'alexa-global-rank': function(content) {
+  'alexa-global-rank': function(content) {
```
Comment (vpodk) [inline tests/testranks.js]: Please remove extra space from the end of line.

File: tests/testtags.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      }
+    }
+  }
+  return true;   
+}
+
+// Export for phantomjs.
+window.testtags = testTags;
+
+
```
Comment (vpodk) [inline tests/testtags.js]: Extra blank line.

File: tests/testtags.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+/**
+ * @fileoverview Defines tests for deprecated tags checker.
+ */
+
+/** @const {!RegExp} */ var PATTERN = /^\d*$/;
+
+/**
+ * @return {boolean} Returns "true" if test failed.
+ */
+function testTags(){
```
Comment (vpodk) [inline tests/testtags.js]: Missed space between `)` and `{`.

File: tests/testtags.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  /** @type {Element} */ 
+  var container = document.getElementById('content-tags-container');
+  /** @type {Element} */
+  var element = container.getElementsByClassName('true')[0];
```
Comment (vpodk) [inline tests/testtags.js]: Check if `container` is exists.

File: tests/testtags.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  /** @type {number} */
+  var message = container.getElementsByClassName('pass')[0].textContent.length;
+  /** @type {Element} */ var table = document.getElementById('tags-data-table');
+  /** @type {Element} */ var nodes = table.getElementsByTagName('tr');
```
Comment (vpodk) [inline tests/testtags.js]: Check if `table` is exists.

File: tests/testtags.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  /** @type {number} */ var length = nodes.length;
+  /** @type {NodeList} */ var th;
```
Comment (vpodk) [inline tests/testtags.js]: Please rename `th` to `cells`.

**Comment by vpodk** [comment]
Please run [jslint.sh](https://github.com/PagePortrait/Report/blob/master/build/jslint.sh) before every commit.

**Comment by AlbaMontis** [comment]
Fixed. Now testtag fails because there are no its widget on pageportrait.

File: tests/testtags.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+ * @return {boolean} Returns "true" if test failed.
+ */
+function testTags() {
+  /** @type {Element} */ 
+  var container = document.getElementById('content-tags-container');
+  /** @type {Element} */
+  var element = container && container.getElementsByClassName('true')[0];
+  /** @type {number} */
+  var message = container && 
+      container.getElementsByClassName('pass')[0].textContent.length;
```
Comment (vpodk) [inline tests/testtags.js]: Check if `container.getElementsByClassName('pass')[0]` is not `undefined`.

File: tests/testtags.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  /** @type {Element} */ var table = document.getElementById('tags-data-table');
+  /** @type {Element} */ var nodes = table && table.getElementsByTagName('tr');
+  /** @type {number} */ var length = nodes && nodes.length;
+  /** @type {NodeList} */ var cells;
+  /** @type {string} */ var txt;
+  /** @type {number} */ var num;
+  /** @type {number} */ var i = 0;
```
Comment (vpodk) [inline tests/testtags.js]: Please put variables in following order:
- variables with values;
- variables w/o values;

File: tests/testtags.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+ */
+function testTags() {
+  /** @type {Element} */
+  var container = document.getElementById('content-tags-container');
+  /** @type {Element} */
+  var element = container && container.getElementsByClassName('true')[0];
+  /** @type {Element} */
+  var passElement = container && container.getElementsByClassName('pass')[0];
+  /** @type {number} */
+  var message = passElement && passElement.textContent.length;
```
Comment (vpodk) [inline tests/testtags.js]: `textContent.trim()` instead of `textContent.length`

**Pull request merged**